### PR TITLE
Automate release, use docker's build actions

### PR DESCRIPTION
**Pull request by burmanm**
Use Docker actions to build and push the docker images, add automated release process when a tag is pushed

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -36,11 +36,35 @@ jobs:
...
+            ${{ runner.os }}-buildx-
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_PASSWORD }}
+          password: ${{ secrets.DOCKER_USERNAME }}
+      - name: Set git short SHA
+        id: vars
+        run: |
+         echo ::set-output name=sha_short::$(git rev-parse --short=8 ${{ github.sha }})
```
Comment (jsanda) [inline .github/workflows/build.yml:62]: Is 8 enough? I use 12 in the Makefile.

**Review by jsanda — CHANGES_REQUESTED** [review]
I think I understand everything here but can you provide a brief summary in case I missed something.

File: .github/workflows/build.yml (hunk 1 above)
Comment (burmanm) [reply .github/workflows/build.yml:62]: Default is 7 (65k unique versions).

**Comment by burmanm** [comment]
Short version of the changes:

* Uses docker's maintained github action to build the docker images instead of makefile's docker-push / docker-build
* In a PR, run only the unit tests
* On pushes to master, also create a docker image and push it with the commit's id as the revision name
* All of these docker builds use caching to speed up the image creation
* On tagging (when using v*), it creates a docker image that uses the tag name as the revision (so: k8ssandra/reaper-operator:v0.0.1) as well as creates a new Github release with the same name (v0.0.1).

File: .github/workflows/build.yml (hunk 1 above)
Comment (jsanda) [reply .github/workflows/build.yml:62]: 👍

**Review by jsanda — APPROVED** [review]
Fantastic! Thanks for doing this 😄

**Pull request merged**